src/webrtc/remoteCallFeed.ts

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { logger } from "../logger";
+import { CallEvent, CallState, MatrixCall } from "./call";
+import { SDPStreamMetadataObject, SDPStreamMetadataPurpose } from "./callEventTypes";
+import { CallFeed, CallFeedEvent, ICallFeedOpts } from "./callFeed";
+import { RemoteCallTrack } from "./remoteCallTrack";
+
+export interface RemoteCallFeedOpts extends ICallFeedOpts {
+    streamId: string;
+    metadata?: SDPStreamMetadataObject;
+    call: MatrixCall;
+
+    /**
+     * @deprecated addTransceiver() should be used instead
+     */
+    stream?: MediaStream;
+}
+
+export class RemoteCallFeed extends CallFeed {
+    private _connected = false;
+    private _metadata?: SDPStreamMetadataObject;
+
+    protected _tracks: RemoteCallTrack[] = [];
+    protected call: MatrixCall;
+    protected _stream: MediaStream;
+
+    public readonly streamId: string;
+    public readonly isLocal = false;
+    public readonly isRemote = true;
+
+    public constructor(opts: RemoteCallFeedOpts) {
+        super(opts);
+
+        if (!opts.metadata && opts.call.opponentSupportsSDPStreamMetadata()) {
+            throw new Error(
+                "Cannot create RemoteCallFeed without metadata if the opponents supports sending sdp_stream_metadata",
+            );
+        }
+
+        this.streamId = opts.streamId;
```

Comment:
Is this also the same as the webrtc stream ID or different? If it's the same, why track it separately?

Reply:
In this specific case, the reason is we create the `RemoteCallFeed` based on the metadata and give it the `streamId` as specified in the metadata and only then add tracks/transceivers to it based on the `streamId` it has as a prop